project/apps/collaboration-service/src/adapters/db/collaboration.supabase.ts

```diff
@@ -312,84 +317,73 @@ export class CollaborationSupabase implements CollaborationRepository {
     return selectedQuestionId;
   }
 
-  async fetchCollabInfo(collabId: string): Promise<CollabInfoDto> {
+  async fetchCollabInfo(collabId: string): Promise<CollabInfoDto | null> {
     const collab = await this.findById(collabId);
     if (!collab) {
-      throw new NotFoundException(
-        `Collaboration with id ${collabId} not found`,
-      );
+      return null;
     }
 
     const { question_id, user1_id, user2_id } = collab;
-    try {
```

Comment:
removed try-catch block here, was originally resulting in throwing within try and catching thrown error in catch